src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java

```diff
@@ -0,0 +1,110 @@
+package eu.elixir.ega.ebi.it.filedatabase;
+
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+
+import com.jayway.restassured.response.Response;
+
+/**
+ * Class to do the integration test for FileController in ega-data-api-key.
+ * 
+ * @author amohan
+ *
+ */
+public class FileControllerIntgTest extends FileDatabaseBase {
+
+    private final static String FILE_CONTROLLER = "/file/";    
+    
+    /**
+     * Verify the api call /file/{fileId} and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also checks the response fileId.
+     */
+    @Test
+    public void testGetFile() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + fileId);
+        JSONArray jsonArray = new JSONArray(response.body().asString()); 
+        JSONObject jsonObject = jsonArray.getJSONObject(0);
+        
+        response.then().assertThat().statusCode(SC_OK);
+        assertThat(jsonArray.length(), equalTo(1));
+        assertThat(jsonObject.getString("fileId"), equalTo(fileId));
+    }
+    
+    /**
+     * Verify the api call /file/{fileId} and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetFileForZeroId() {

```

Comment:
The same comment as for the non existing dataset

Reply:
Yes, currently it is implemented like this. It returns empty result array if id not found in the database.